From f7d5881497a299688d69cc975bcae18a7d716fb2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lubo=C5=A1=20Lu=C5=88=C3=A1k?= <l.lunak@collabora.com>
Date: Thu, 8 Nov 2018 15:08:48 +0100
Subject: [PATCH] blacklist svMatrix for calc threaded calculation

If a formula contains a matrix, interpreting it will actually result
in modifying it in some ways (e.g. ScInterpreter::PopRangeMatrix() calls
SetErrorInterpreter() on the matrix).
Testcase: rhbz#1645905

Change-Id: I63742449ba0e4e375d6e459dc7bccd7e6b62920e
---
 sc/source/core/tool/token.cxx | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sc/source/core/tool/token.cxx b/sc/source/core/tool/token.cxx
index db03b6a..494d162 100644
--- a/sc/source/core/tool/token.cxx
+++ b/sc/source/core/tool/token.cxx
@@ -1366,6 +1366,8 @@ void ScTokenArray::CheckForThreading( const FormulaToken& r )
         {
             case svExternalDoubleRef:
             case svExternalSingleRef:
+            case svExternalName:
+            case svMatrix:
                 SAL_INFO("sc.core.formulagroup", "opcode ocPush: variable type " << StackVarEnumToString(r.GetType())
                     << " disables threaded calculcation of formula group");
                 mbThreadingEnabled = false;
-- 
2.19.1

